### PR TITLE
[Docs] Add keys tutorial

### PR DESCRIPTION
**Pull request by tcbegley**
## Description

This PR adds a tutorial for manipulating keys in a `TensorDict`.

### CODE REVIEW

File: tutorials/sphinx_tuto/tensordict_shapes.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+# whose entries are views of the corresponding entries in the original
+# :class:`~.TensorDict`.
+
+chunks = tensordict.split([3, 1], dim=1)
+assert chunks[0].batch_size == torch.Size([3, 3])
+assert chunks[1].batch_size == torch.Size([3, 1])
+torch.testing.assert_allclose(chunks[0]["a"], tensordict["a"][:, :-1])
+
+##############################################################################
+# .. note::
```
Comment (vmoens) [inline tutorials/sphinx_tuto/tensordict_shapes.py]: same here 
let's add an empty line below

File: tutorials/sphinx_tuto/tensordict_keys.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+tensordict.set("a", a)
+
+# retrieve the value stored under "a"
+assert tensordict.get("a") is a
+
+###############################################################################
+# Like ``dict``, we can provide a default value to ``get`` that should be returned in
+# case the requested key is not found.
+
+assert tensordict.get("banana", a) is a
```
Comment (vmoens) [inline tutorials/sphinx_tuto/tensordict_keys.py:56]: maybe let's talk about set_default here?
IIRC we used an underscore when dicts don't, we should correct that

File: tutorials/sphinx_tuto/tensordict_keys.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    else:
+        print(f"{key} is a Tensor")
+
+###############################################################################
+# Checking for existence of a key
+# -------------------------------
+# To check if a key exists in a ``TensorDict``, use the ``in`` operator in conjunction
+# with ``.keys()``.
+#
+# .. note::
```
Comment (vmoens) [inline tutorials/sphinx_tuto/tensordict_keys.py:146]: same here

File: tutorials/sphinx_tuto/tensordict_keys.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+assert "a" in tensordict.keys()
+# to check for nested keys, set include_nested=True
+assert ("nested", "a") in tensordict.keys(include_nested=True)
+assert ("nested", "banana") not in tensordict.keys(include_nested=True)
+
+################################################################################
+# Flattening nested keys
+# ----------------------
+# We can flatten a ``TensorDict`` with nested values using the ``.flatten_keys()``
```
Comment (vmoens) [inline tutorials/sphinx_tuto/tensordict_keys.py:159]: Let's add that unflattening is super useful for torch.nn.Modules, bc you can get a tensordict structure that looks like your model

File: tutorials/sphinx_tuto/tensordict_keys.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+print(tensordict, end="\n\n")
+print(tensordict.flatten_keys(separator="."))
+
+##############################################################################
+# Given a ``TensorDict`` that has been flattened, it is possible to unflatten it again
+# with the ``.unflatten_keys()`` method.
+
+flattened_tensordict = tensordict.flatten_keys(separator="-")
+print(flattened_tensordict, end="\n\n")
+print(flattened_tensordict.unflatten_keys(separator="-"))
```
Comment (vmoens) [inline tutorials/sphinx_tuto/tensordict_keys.py:171]: To add: select, exclude, rename_key

File: tutorials/sphinx_tuto/tensordict_keys.py (hunk 2 above)
Comment (tcbegley) [reply tutorials/sphinx_tuto/tensordict_keys.py:56]: I can submit a PR for that.

Comment (tcbegley) [reply tutorials/sphinx_tuto/tensordict_keys.py:56]: https://github.com/pytorch-labs/tensordict/pull/235

File: tutorials/sphinx_tuto/tensordict_keys.py (hunk 5 above)
Comment (tcbegley) [reply tutorials/sphinx_tuto/tensordict_keys.py]: Done!

**Comment by tcbegley** [comment]
Note that I've pre-emptively used `setdefault` rather than `set_default` so the CI might fail until #235 is merged.

**Review by vmoens — APPROVED** [review]
LGTM thanks!

**Pull request merged**